tests/unit/test_unit_smhi.py

```diff
@@ -0,0 +1,249 @@
+"""SMHI unit tests."""
+
+from unittest.mock import MagicMock, patch
+
+import pandas as pd
+import pytest
+from smhi.smhi import SMHI
+
+
+class TestUnitSMHI:
+    """Unit tests for SMHI class."""
+
+    @patch("smhi.metobs.Parameters.__new__")
+    def test_unit_smhi_init(self, mock_parameters):
+        """Unit test for SMHI init method.
+
+        Args:
+            mock_requests_metobs: mock requests metobs object
+        """
+        client = SMHI()
+
+        assert client.parameters
+
+    @pytest.mark.parametrize("parameter", [(None), (1)])
+    @patch("smhi.metobs.Stations.__new__")
+    def test_unit_smhi_get_stations(self, mock_station_data, parameter):
+        """Unit test for SMHI get_stations method.
+
+        Args:
+            parameter: parameter (int)
+        """
+        client = SMHI()
+        assert client.get_stations(parameter)
+
+    @pytest.mark.parametrize("parameter_title", [(None), ("Snöfall")])
+    @patch("smhi.metobs.Stations.__new__")
+    def test_unit_smhi_get_stations_from_title(
+        self, mock_station_data, parameter_title
+    ):
+        """Unit test for SMHI get_stations_from_title method.
+
+        Args:
+            title: title of station
+        """
+        client = SMHI()
+        assert client.get_stations_from_title(parameter_title)
+
+    @pytest.mark.parametrize(
+        "parameter, station, distance", [(8, 180960, None), (8, 180960, 50)]
+    )
+    @patch("smhi.metobs.Stations.__new__")
+    @patch("smhi.metobs.Periods.__new__")
+    @patch("smhi.metobs.Data.__new__")
+    @patch("smhi.smhi.SMHI._interpolate")
+    def test_unit_get_data(
+        self,
+        mock_interpolate,
+        mock_data_data,
+        mock_period_data,
+        mock_station_data,
+        parameter,
+        station,
+        distance,
+    ):
+        mock_interpolate.return_value = "test"
+        client = SMHI()
+        data = client.get_data(parameter, station, distance)
+        assert data == "test"
+
+    @pytest.mark.parametrize(
+        "parameter, city, distance", [(8, "Bengtsfors", None), (8, "Bengtsfors", 50)]
+    )
+    @patch("smhi.metobs.Stations.__new__")
+    @patch("smhi.metobs.Periods.__new__")
+    @patch("smhi.metobs.Data.__new__")
+    @patch("smhi.smhi.SMHI._find_stations_by_city")
+    @patch("smhi.smhi.SMHI._interpolate")
+    def test_unit_get_data_by_city(
+        self,
+        mock_interpolate,
+        mock_find_stations_by_city,
+        mock_data_data,
+        mock_period_data,
+        mock_station_data,
+        parameter,
+        city,
+        distance,
+    ):
+        mock_interpolate.return_value = "test"
+        client = SMHI()
+        data = client.get_data_by_city(parameter, city, distance)
+        assert data == "test"
+
+    distanceresponse = MagicMock()
+    distanceresponse.km = 0
+
+    @pytest.mark.parametrize(
+        "latitude, longitude,dist",
+        [
+            (
+                59,
+                17,
+                0,
```

Comment:
```suggestion
                0
```

Remove comma to put on one line.

Reply:
This is the formatting I get when automatically saving after writing everything on one line, are you sure removing this comma is the right thing to do? Why is the automatic formatting adding it and why doesn't it cause lint issues?

Reply:
My formatter returns this as one line

```
@pytest.mark.parametrize("latitude, longitude,dist", [(59, 17, 0), (59.4, 17, 30)])
```

Reply:
Black and ruff will usually try to do this, at least for me. It will format this

```python
def fun(self,):
    pass
```

into 

```python
def fun(
    self,
):
    pass
```

If you don't explicitly remove the comma after self.

I have never experienced it adding the comma if it isn't needed, but maybe you have a different allowed max lie length. It isn't important, it's just taking up a lot of space, sometimes for no reason.

Reply:
Ok, thanks for clearing that up for me. 👍